plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/ConnectionFactory.java

```diff
@@ -47,7 +48,7 @@ public Connection create(String url) {
 	public Connection create(String url, ICredentialsPrompter credentialsPrompter) {
 		try {
 			return new Connection(url, 
-					new AuthorizationClient(), 
+					new AuthorizationClientFactory().create(), 
```

Comment:
Add factory here so we could have that layer of indirection where the ConnectionFactory doesnt depend on AuthorizationClient but IAuthorizationClient